Ttonev/hr portal

tishko0 opened the pull request.

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+  }
+
+  async fetchData() {
+    try {
```
Can you move this to a separate service file so that it would be similar as the other samples

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+
+  async fetchData() {
+    try {
+      const response = await fetch('https://staging7.infragistics.com/grid-examples-data/data/hr/hr.json');
```
move that url to a constant var

tishko0 replied:
moved

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+  }
+
+  clearSorting() {
+    const treeGrid = this.shadowRoot?.getElementById('treeGrid') as IgcTreeGridComponent;
```
You can use the @query decorator to get a referance to the grid

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+  }
+  
+  public avatarTemplate = (ctx: IgcCellTemplateContext) => {	
+    let row = ctx.cell.row;
```
isnt this the same as `ctx.cell.value`? why get the value from the row. Same as all other body templates

tishko0 replied:
no, the value is the name, the avatar is looking for picture path which is another property. Same goes for other templates. The contact template takes value from 3 fields.

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+  render() {
+    return html`
+      <link rel="stylesheet" href="node_modules/igniteui-webcomponents-grids/grids/themes/light/fluent.css" />
+      <igc-tree-grid
```
There is a tree grid selector that is redundant

tishko0 replied:
removed

mddragnev commented on projects/hr-portal/index.html:
```diff
@@ -7,8 +7,12 @@
     <script type="module" src="./src/hr-portal.ts"></script>
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
+    <link href="https://fonts.googleapis.com/css2?family=Open+Sans:wght@300;400;500;700&display=swap" rel="stylesheet">
+    <link rel="stylesheet" href="./node_modules/igniteui-webcomponents/themes/light/fluent.css" />
```
Do you need this?

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
-      color: #535bf2;
-    }
+  igc-tree-grid::part(row) {
+      background-color: #ffffff(--row-even-background);
```
I think this is not used. Also, it is not a correct value for `background-color`

tishko0 replied:
it applies different coloring for even rows, same as in angular.  Not very visible on some displays but it does apply.

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,195 @@
+      display: block;
       margin: 0 auto;
-      padding: 2rem;
+      align-items: center;
```
If this is gonna be a block container, then you dont need `align-items`, `justify-content` and `gap`. Probably, wont need the `margin` and `text-allign` as well

tishko0 replied:
removed the unnecessary properties

mddragnev commented on projects/hr-portal/src/index.css:
```diff
@@ -0,0 +1,17 @@
+  }
+  
+  app-hr-portal {
+    flex: 1;
```
These override the `:host` styles that you set in the `hr-portal.ts` file. If you dont need this, please remove it. Also, why flex on a host that has only 1 element?

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
Apply some formatting to the file as well

tishko0 replied:
done

tishko0 commented:
> * There are some minor difference between the angular toolbar and this one.
>   Here's a comparison, angular on the right, wc on the left.
>   ![image](https://private-user-images.githubusercontent.com/10397980/426070613-6c3b955f-42f8-4cf8-81c0-4d128da23d04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzA2MTMtNmMzYjk1NWYtNDJmOC00Y2Y4LTgxYzAtNGQxMjhkYTIzZDA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4ZjkwMzkwNzVkNzAyOWRmNGJlYzdjYjhkZTEwMjJhOWI3YTM1OTEwNTgxM2IzMWZjMmZkNmQ5ZDdkMmQ0YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JrKVTAIRXCXPmWeZssXWQwZQggSXUoDf5sV69abU018)
>   Notice that:
>   
>   * Title "HR Portal" is not aligned to the left in the wc one.
>   * Button text in the toolbar is darker in the Angular one, compared to the wc one.
> * Also after sorting a column, in angular a new button for clearing sort shows. In wc it does not show:
> 
> ![image](https://private-user-images.githubusercontent.com/10397980/426077881-3d278031-dcb7-4ec8-90f1-43e493792509.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzc4ODEtM2QyNzgwMzEtZGNiNy00ZWM4LTkwZjEtNDNlNDkzNzkyNTA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjN2VhN2E3MTMyNzdkMDBhOTI1NmQ3YjE3ZDRjMzA2OWM3NmY3MTBjYTQ2NGEzYmJlNmY5MWQzZWViYjE5ZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FjUJqNglzCGtG4M9TDQ3rzaSKcriF7vRNHHhV-XCJb0)
> 
> * If we compare both the angular and wc sample with the [figma design](https://www.figma.com/design/FLpO8ARKvgMzgE5k8QvNPM/Grid-Landing-Page-%26-Samples?node-id=265-142922&t=D7G6bfdjm1B7z1KQ-0):
>   
>   * Title in design is "Actions", while in the sample it is "HR Portal"
>   * Column "Name" is initially sorted in the design, but not in the samples.
>   
>   Either the samples or the design should be updated so that there are no differences.
> * Noticed that there are also some console.logs left over. Those should be removed:
>   ![image](https://private-user-images.githubusercontent.com/10397980/426073503-1e7907bd-4014-4b9e-8786-a7e7c04fcc61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzM1MDMtMWU3OTA3YmQtNDAxNC00YjllLTg3ODYtYTdlN2MwNGZjYzYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhZjI5MTk0OWE5NmM4MThlMWZhOTI1YWY2MWMxNzE3YTdhMjVjMTI1NGQyYjZiMzM1ZDA0Mzg2YWRmZDA4OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aFk1On7BUKPiPlBWOc-bCIUAIMf_IxF0-rm4-7ZE5B0)

Added the sorting button and changed the title to actions. Regarding text color, I have not set any formatting to it in both samples, so i guess it comes with difference in themes between WC in Angular. We can comment further if this needs to be changed. Regarding the toolbar title alignment, it does not align on the right side for me. Maybe someone else can test it too?

tishko0 commented on projects/hr-portal/public/images/Icons.ts:
moved it

tishko0 commented on projects/hr-portal/src/index.css:
```diff
@@ -0,0 +1,17 @@
+  
+  body {
+    display: flex;
+    font-family: 'Helvetica', sans-serif;
```
As above, fine with scrollbar, wont be consistent with angular. Regarding typograhy, not sure what var i should use for helvetica. The default value changes the font family to different one.

mddragnev replied:
It should be consistent with angular. You have those clases in angular. 
![Screenshot 2025-03-26 at 16 19 06](https://github.com/user-attachments/assets/7cb8f530-c696-4c39-b5d7-102380170153)
Also, the font family is coming from the theme. There is a Helvetica font. You should use that variable

mddragnev commented on projects/hr-portal/src/index.css:
```diff
@@ -0,0 +1,17 @@
+  }
+  
+  body {
+    display: flex;
```
You can set width/height to the parent containers since block elements occupies the whole space when sizes set in percentages

MayaKirova commented:
> > * There are some minor difference between the angular toolbar and this one.
> >   Here's a comparison, angular on the right, wc on the left.
> >   ![image](https://private-user-images.githubusercontent.com/10397980/426070613-6c3b955f-42f8-4cf8-81c0-4d128da23d04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzA2MTMtNmMzYjk1NWYtNDJmOC00Y2Y4LTgxYzAtNGQxMjhkYTIzZDA0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4ZjkwMzkwNzVkNzAyOWRmNGJlYzdjYjhkZTEwMjJhOWI3YTM1OTEwNTgxM2IzMWZjMmZkNmQ5ZDdkMmQ0YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.JrKVTAIRXCXPmWeZssXWQwZQggSXUoDf5sV69abU018)
> >   Notice that:
> >   
> >   * Title "HR Portal" is not aligned to the left in the wc one.
> >   * Button text in the toolbar is darker in the Angular one, compared to the wc one.
> > * Also after sorting a column, in angular a new button for clearing sort shows. In wc it does not show:
> > 
> > ![image](https://private-user-images.githubusercontent.com/10397980/426077881-3d278031-dcb7-4ec8-90f1-43e493792509.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzc4ODEtM2QyNzgwMzEtZGNiNy00ZWM4LTkwZjEtNDNlNDkzNzkyNTA5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjN2VhN2E3MTMyNzdkMDBhOTI1NmQ3YjE3ZDRjMzA2OWM3NmY3MTBjYTQ2NGEzYmJlNmY5MWQzZWViYjE5ZjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.FjUJqNglzCGtG4M9TDQ3rzaSKcriF7vRNHHhV-XCJb0)
> > 
> > * If we compare both the angular and wc sample with the [figma design](https://www.figma.com/design/FLpO8ARKvgMzgE5k8QvNPM/Grid-Landing-Page-%26-Samples?node-id=265-142922&t=D7G6bfdjm1B7z1KQ-0):
> >   
> >   * Title in design is "Actions", while in the sample it is "HR Portal"
> >   * Column "Name" is initially sorted in the design, but not in the samples.
> >   
> >   Either the samples or the design should be updated so that there are no differences.
> > * Noticed that there are also some console.logs left over. Those should be removed:
> >   ![image](https://private-user-images.githubusercontent.com/10397980/426073503-1e7907bd-4014-4b9e-8786-a7e7c04fcc61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODAzNDksIm5iZiI6MTc0Mjk4MDA0OSwicGF0aCI6Ii8xMDM5Nzk4MC80MjYwNzM1MDMtMWU3OTA3YmQtNDAxNC00YjllLTg3ODYtYTdlN2MwNGZjYzYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5MDcyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhZjI5MTk0OWE5NmM4MThlMWZhOTI1YWY2MWMxNzE3YTdhMjVjMTI1NGQyYjZiMzM1ZDA0Mzg2YWRmZDA4OTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aFk1On7BUKPiPlBWOc-bCIUAIMf_IxF0-rm4-7ZE5B0)
> 
> Added the sorting button and changed the title to actions. Regarding text color, I have not set any formatting to it in both samples, so i guess it comes with difference in themes between WC in Angular. We can comment further if this needs to be changed. Regarding the toolbar title alignment, it does not align on the right side for me. Maybe someone else can test it too?


- Regarding the styling of the text in the buttons, the same does not happen in the web component samples. If you get one of the samples, lets say this one : https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/grids/grid/toolbar
And change the theme to fluent, for example: https://codesandbox.io/p/sandbox/solitary-bash-ql6797

Then the text has the same styles as in angular:

![image](https://github.com/user-attachments/assets/a7068346-1201-43f7-9a12-363077590c9b)

So it seems to be some issue with the sample or the configuration in lit with the theming.

- Regarding the text of the title, I assume the design is outdated and the one in angular is the correct one. @dkamburov probably has more information.

- Another issue I noticed with the latest changes is that the sample does not flex to the full 100% width:
![image](https://github.com/user-attachments/assets/f623c920-a855-4ee4-abbb-f4d670801bb1)
It's probably missing a flex-grow or something to grow the content to the page size.

- Also, the "Clear Sort" button is there, but for some reason the icon is outside of it and it doesn't look good when you hover it:
![image](https://github.com/user-attachments/assets/a554568e-180f-433f-8b12-b947802a3e2b)

It should be inside, like in angular:
![image](https://github.com/user-attachments/assets/0b8c0da1-b392-4a7d-ab12-9d585721ba54)

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,265 @@
+  handleSortingChanged() {
+    if (this.treeGrid) {
+      this.isSorted = this.treeGrid.sortingExpressions.length > 0;
+      console.log(
```
There is a console log here

tishko0 replied:
removed

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,265 @@
+  render() {
+    return html`
+      <link rel="stylesheet" href="node_modules/igniteui-webcomponents-grids/grids/themes/light/fluent.css" />
+      <div class="ig-typography rootDiv">
```
Why do you need that div container?

tishko0 replied:
the typography does not apply otherwise even tho its already applied to the body

mddragnev replied:
It does on my machine. Are you sure it doesnt?

tishko0 replied:
yes, it came as suggestion from @skrustev  as he said it didn't on his as well

mddragnev commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,265 @@
+          <igc-icon-button
+            collection="hr-icons"
+            name="mail"
+            class="small"
```
You can move the class `small` to the parent div so that you dont apply the same class over again on all elements

tishko0 replied:
moved to the parent div

tishko0 commented on projects/hr-portal/src/hr-portal.ts:
```diff
@@ -1,83 +1,265 @@
 
-    .card {
-      padding: 2em;
+    .rootDiv {
```
so long the div stays due to the above
